src/sentry/rules/actions/utils.py

```diff
@@ -0,0 +1,116 @@
+from collections import defaultdict
+from typing import Any, DefaultDict
+
+from sentry.api.serializers.models.rule import generate_rule_label
+from sentry.models.environment import Environment
+from sentry.models.rule import Rule
+
+
+def get_updated_rule_data(rule: Rule) -> dict[str, Any]:
+    rule_data = dict(rule.data)
+    if rule.environment_id:
+        rule_data["environment_id"] = rule.environment_id
+    if rule.owner:
+        rule_data["owner"] = rule.owner
+    rule_data["label"] = rule.label
+    return rule_data
+
+
+def check_value_changed(
+    rule_data: dict[str, Any], rule_data_before: dict[str, Any], key: str, word: str
+) -> str:
+    if rule_data.get(key) != rule_data_before.get(key):
+        old_value = rule_data_before.get(key)
+        new_value = rule_data.get(key)
+        return f"Changed {word} from *{old_value}* to *{new_value}*"
+    return ""
+
+
+def check_added_or_removed(
```

Comment:
ok tell me if this function is too confusing, this was hard to write to compare new to told and also old to new and to take in stuff like "actions" but write "action"

Reply:
Hmm maybe a name like `generate_diff_labels` would be good?